lib/deck.rb

```diff
@@ -0,0 +1,45 @@
+class Deck 
+  attr_reader :high_ranking_card, :percent_high_ranking
+  attr_accessor :cards, :rank_of_card_at
```

Comment:
To speak to your question about attr_reader vs attr_accessor, `attr_accessor` is only necessary if you want the ability to **read** as well as **write** the value of the instance variable it exposes, from outside of the class. Usually, an `attr_reader` is sufficient.
If you haven't already, try running your tests and `war_or_peace_runner.rb` with all of these defined as `attr_reader` and see what happens.